core/src/main/java/com/sequenceiq/cloudbreak/service/image/userdata/UserDataBuilder.java

```diff
@@ -62,9 +64,26 @@ private String build(InstanceGroupType type, Platform cloudPlatform, byte[] cbSs
         model.put("saltBootPassword", saltBootPassword);
         model.put("cbCert", cbCert);
         CcmParameters.addToTemplateModel(ccmParameters, model);
+        extendModelWithProxyParams(type, proxyConfig, model);
         return build(model);
     }
 
+    private void extendModelWithProxyParams(InstanceGroupType type, ProxyConfig proxyConfig, Map<String, Object> model) {
+        if (type == InstanceGroupType.GATEWAY && proxyConfig != null) {
+            model.put("proxyEnabled", Boolean.TRUE);
+            model.put("proxyHost", proxyConfig.getServerHost());
+            model.put("proxyPort", proxyConfig.getServerPort().toString());
+            if (!StringUtils.isEmpty(proxyConfig.getUserName())) {
+                model.put("proxyUser", proxyConfig.getUserName());
+            }
+            if (!StringUtils.isEmpty(proxyConfig.getPassword())) {
+                model.put("proxyPassword", proxyConfig.getPassword());
+            }
```

Comment:
you should add if all has value, so you could use `StringUtils#isNoneBlank` here